File: python/tests/dashboard/test_questions.py
from .base_test import DashboardBaseTest


class TestQuestions(DashboardBaseTest):

    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.sample_dashboard.datasources.link(cls.sample_datasource_id)

    def test_list(self):
        self.assertShouldRaise(
            None,
            lambda: self.onvoSDK.questions.list(self.sample_dashboard_id),
        )

    def test_delete(self):
        sample_question_id = self.onvoSDK.questions.create(
            self.sample_dashboard_id, "How many rows does the data have?"
        )[
            "id"
        ]  # skipping create test
        self.assertShouldRaise(
            None,
            lambda: self.onvoSDK.questions.delete(sample_question_id),
        )

File: ruby/test/dashboard/test_questions.rb
# frozen_string_literal: true

require_relative './base_dashboard_template'

# All tests related to Onvo's dashboard question endpoints
class QuestionTest < BaseDashboardTemplate
  def before_all
    super
    @dashboard.datasources.link(@sample_datasource_id)
  end

  def test_list_dashboard_questions
    assert_silent { @onvo.questions.list(@sample_dashboard_id) }
  end

  def test_delete_dashboard_question
    sample_question_id = @onvo.questions.create(@sample_dashboard_id, 'How many rows does the data have?')['id'] # skipping test create
    assert_silent { @onvo.questions.delete(sample_question_id) }
  end
end
